google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -1693,15 +1689,27 @@ public PooledSession get() {
       }
     }
 
-    private PooledSession pollUninterruptiblyWithTimeout(long timeoutMillis) {
+    private PooledSession pollUninterruptiblyWithTimeout(
+        long timeoutMillis, Duration acquireSessionTimeout) {
       boolean interrupted = false;
       try {
         while (true) {
           try {
-            return waiter.get(timeoutMillis, TimeUnit.MILLISECONDS);
+            // TODO refactor this code since eventually acquireSessionTimeout will always have a
+            // default value and won't be null
+            return acquireSessionTimeout == null
+                ? waiter.get(timeoutMillis, TimeUnit.MILLISECONDS)
+                : waiter.get(acquireSessionTimeout.toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
We should add a check to the `SpannerOptions$Builder#setAcquireSessionTimeout` method that ensures that no one ever sets a timeout that is bigger than can be converted to milliseconds. Otherwise, this method could return an `ArithmeticException`.

Reply:
Added a validation.